### PR TITLE
First Pass at Basic Learning Tools

### PR DESCRIPTION
**Pull request by elmq0022**
So there's not much in the way of CSS and stuff.   Hopefully it's a reasonable first pass at the functionality.

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -41,7 +43,13 @@ class SignUpView(generic.CreateView):
 class Home(LoginRequiredMixin, TemplateView):
     template_name = 'accounts/home.html'
 
+    def top_characters(self):
+        return Count.objects.filter(count_type=Count.CHARACTER).all().order_by('-count')[:5] 
+
+    def top_words(self):
+        return Count.objects.filter(count_type=Count.WORD).all().order_by('-count')[:5] 
+
     def dispatch(self, request, *args, **kwargs):
```
Comment (jordanbettis) [inline accounts/views.py:52]: Is there a reason why you're not using the `login_required` decorator?

```python
    @login_required(login_url=reverse('hanyu:home'))
    def dispatch(self, *args, **kwargs):
        return super().dispatch(*args, **kwargs)
````

File: learning_tools/models.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    LEARNING_STATUS = (
+        (UNLEARNED, 'unlearned'),
+        (ACQUIRING, 'acquiring'),
+        (LEARNED, 'learned'),
+    )
+    entry = models.ForeignKey(Entry)
+    user = models.ForeignKey(User)
+    learning_status = models.CharField(max_length=2,
+                                       choices=LEARNING_STATUS,
+                                       default='UN',
```
Comment (jordanbettis) [inline learning_tools/models.py]: shouldn't this be `default=UNLEARNED`?

**Review by jordanbettis — APPROVED** [review]
Looks good!

File: learning_tools/models.py (hunk 2 above)
Comment (elmq0022) [reply learning_tools/models.py]: Agreed, I will update that.

File: accounts/views.py (hunk 1 above)
Comment (elmq0022) [reply accounts/views.py:52]: In this case, isn't using @login_required equivalent to Django's LoginRequiredMixin [mixins.py](https://github.com/django/django/blob/301de774c21d055e9e5a7073e5bffdb52bc71079/django/contrib/auth/mixins.py) that I inherited from in the class?  Is there a general preference over one method or the other?  Thanks!

Comment (jordanbettis) [reply accounts/views.py:52]: Heh. I didn't know about that mixin. Must be a newish thing.

Using either the mixin or the decorator, the below code is redundant and should be removed.

**Review by jordanbettis — APPROVED** [review]

**Pull request merged**